### PR TITLE
decode cookie key and value when parsing

### PR DESCRIPTION
**Pull request by ghost**
Parsing a cookie then stringifying it (as HttpApps.CookieJar does) causes double-encoding. This fixes that.

CookieJar was also incorrectly sending cookie metadata in the request, fixed by only sending two parameters to stringify from CookieJar.

CookieJar was also duplicating cookie values.  After debugging I figured out that it was using `Array.prototype.map` to iterate over a set of values, and _never using the iterator_.  I removed that unnecessary loop.

### CODE REVIEW

**Comment by kriskowal** [comment]
@stuk, I could use your eyes on this.

File: http-cookie.js [hunk 1]
```diff
@@ -17,8 +17,8 @@ exports.parse = function (cookie, date) {
     var parsed = {};
     var terms = cookie.split(/[;,]/g);
     var keyValue = terms.shift().split("=");
-    parsed.key = keyValue[0];
-    parsed.value = keyValue[1];
+    parsed.key = decodeURIComponent(keyValue[0]);
+    parsed.value = decodeURIComponent(keyValue[1]);
```
Comment (Stuk) [inline http-cookie.js:21]: This change looks good, matches up with `stringify`

File: http-apps/cookie.js [hunk 2]
```diff
@@ -68,8 +64,7 @@ exports.CookieJar = function (app) {
                 .map(function (cookie) {
                     return Cookie.stringify(
                         cookie.key,
-                        cookie.value,
-                        cookie
+                        cookie.value
```
Comment (Stuk) [inline http-apps/cookie.js:67]: This change also looks good. The cookie header should only send key=value

File: http-apps/cookie.js [hunk 3]
```diff
@@ -14,53 +14,49 @@ exports.CookieJar = function (app) {
 
         var now = new Date();
 
-        var requestCookies = concat(hosts.map(function (host) {
```
Comment (Stuk) [inline http-apps/cookie.js:17]: In the [no-whitespace diff](https://github.com/kriskowal/q-io/pull/128/files?w=0) this change is small. Reading through I can't tell if the logic is correct, I would need to step through this and read the cookie spec.

Removing the `.map` does leave the `hosts` variable unused.

**Pull request merged**